test/controllers/works_controller_test.rb

```diff
@@ -0,0 +1,46 @@
+require "test_helper"
+
+describe WorksController do
+  let(:work) { works :fitness }
+
+  it "gets index" do
+    get works_url
+    value(response).must_be :success?
+  end
+
+  it "gets new" do
+    get new_work_url
+    value(response).must_be :success?
+  end
+
+  it "creates work" do
+    expect {
+      post works_url, params: { work: { category: work.category, creator: work.creator, description: work.description, publication_year: work.publication_year, title: work.title } }
+    }.must_change "Work.count"
+
+    must_redirect_to work_path(Work.last)
+  end
+
+  it "shows work" do
+    get work_url(work)
+    value(response).must_be :success?
+  end
+
+  it "gets edit" do
+    get edit_work_url(work)
+    value(response).must_be :success?
+  end
+
+  it "updates work" do
+    patch work_url(work), params: { work: { category: work.category, creator: work.creator, description: work.description, publication_year: work.publication_year, title: work.title } }
+    must_redirect_to work_path(work)
+  end
+
+  it "destroys work" do
+    expect {
+      delete work_url(work)
+    }.must_change "Work.count", -1
+
+    must_redirect_to works_path
```

Comment:
There's a lot of interesting test cases for your custom `Work` methods missing here. For `top?`, I would wonder:

- What happens if there are no works?
- What happens if there are works but no votes?
- What happens if two works have the same number of votes?

Similarly for your 'sample' method, I would ask:

- What if there are no works of that category?
- What if there are less than 10 works?
- What if there's a tie for last place, e.g. works 9, 10 and 11 all have 0 votes?